### DISCUSSION TITLE
Sending soap call to wsdl getting A SOAP 1.2 message is not valid when sent to a SOAP 1.1 only endpoint

### DISCUSSION THREAD

**Discussion by mnouwens** [quarkiverse/quarkus-cxf#309] [Q&A]
We implemented a CxfClient.

To wiremock it works as expected but on the real endpoint we get:

A SOAP 1.2 message is not valid when sent to a SOAP 1.1 only endpoint

Can we tell the client to send 1.1 format?

**Comment by shumonsharif** [comment]
Hi @mnouwens You can try setting the following in your `application.properties` file:

```properties
quarkus.cxf.client.key.soap-binding=http://schemas.xmlsoap.org/wsdl/soap/http
```

For reference, you can take a look [here](https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/dev/properties.html#quarkus.cxf.endpoint.key.soap-binding) for the other possible URL values; though looks like the soap-binding docs could use some polishing.

**Reply by mnouwens** [reply]
Hi, Al already tried that. Did not help.

Any other suggestions?

Basicly it means that it's sending a soap 1.2 message to an 1.1 endpoind. But 1.1 is the default in quarkus. So that should already be ok. Is there somthing i can do in the wsdl or code generation from the wsdl. On the client. I have no control over the endpoint?

Marteijn

**Reply by shumonsharif** [reply]
Hey Marteijn, you're correct, the default soap binding on the client is 1.1.  I was wondering why you were trying to "tell the client to send 1.1 format", I assumed perhaps you had the binding being specified and overridden somewhere else.  Also, I've seen this same error in the past where a soap 1.2 message gets **returned as a response from an endpoint**, and the client is configured for 1.1.

While it may be possible to convert the WSDL from 1.2 to 1.1; it is somewhat unlikely that the WSDL or the code generation is the issue ... in case you didn't try already, would you please set your binding value to the SOAP 1.2 URL: `http://www.w3.org/2003/05/soap/bindings/HTTP/` and let us know the results?

**Reply by mnouwens** [reply]
I have tried:

#quarkus.cxf.client.key.soap-binding=http://schemas.xmlsoap.org/wsdl/soap/http
#quarkus.cxf.client.key.soap-binding=http://schemas.xmlsoap.org/wsdl/soap/http
quarkus.cxf.client.key.soap-binding=http://www.w3.org/2003/05/soap/bindings/HTTP/
#quarkus.cxf.endpoint.key.soap-binding=http://www.w3.org/2003/05/soap/bindings/HTTP/
#quarkus.cxf.client."domibus-client".soap-binding=http://www.w3.org/2003/05/soap/bindings/HTTP/

These all do not work

Ik will add an intercepter to see what is being send.

**Reply by shumonsharif** [reply]
Hi Marteijn, only the last "domibus-client" property in your list is applicable; apologies I wasn't explicit about replacing the word 'key' with the actual client or endpoint name.  

I spent some time putting together a sample maven project that may possibly be helpful for you.  There's a unit test class named `CalculatorSoapTest` that demonstrates various scenarios and potential exceptions that may get thrown - please have a look, and let me know if you have any questions.  

Hopefully the unit test provides some context for you to be able to address your issue.

[quarkus-cxf-soap11-soap12.zip](https://github.com/quarkiverse/quarkus-cxf/files/7500187/quarkus-cxf-soap11-soap12.zip)

**Reply by antoniofloinditex** [reply]
my lfe is a fucking shit since this error

**Reply by mnouwens** [reply]
It's working for me now.

I have:
quarkus.cxf.client.key.soap-binding=http://schemas.xmlsoap.org/wsdl/soap/http
quarkus.cxf.client."domibus-client".out-interceptors=org.apache.cxf.ext.logging.LoggingOutInterceptor
quarkus.cxf.client."domibus-client".in-interceptors=org.apache.cxf.ext.logging.LoggingInInterceptor
quarkus.cxf.client."domibus-client".service-interface=eu.domibus.submission.SubmissionEnvelopeInterface
quarkus.cxf.client."domibus-client".client-endpoint-url=https://service.eu/domibus/services/messageContainer

Nothing more 

Hope this help. But if you have  ldap authentication use:
```
 <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-elytron-security-ldap</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>org.wildfly.security</groupId>
                    <artifactId>wildfly-elytron-client</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```
Don't remember exaclty it does not work properly unles i exclude that dep.
